Updated to match fabric version

Stereowalker opened the pull request.

Stereowalker committed:
Updated to match fabric version

tyra314 commented on src/main/java/hunternif/mc/impl/atlas/client/gui/GuiBookmarkButton.java:
```diff
@@ -56,12 +57,14 @@ public void render(MatrixStack matrices, int mouseX, int mouseY, float partialTi
 		int u = colorIndex * WIDTH;
 		int v = isMouseOver || isSelected() ? 0 : HEIGHT;
-		AtlasRenderHelper.drawTexturedRect(matrices, Textures.BOOKMARKS, getGuiX(), getGuiY(), u, v, WIDTH, HEIGHT, IMAGE_WIDTH, IMAGE_HEIGHT);
+//		AtlasRenderHelper.drawTexturedRect(matrices, Textures.BOOKMARKS, getGuiX(), getGuiY(), u, v, WIDTH, HEIGHT, IMAGE_WIDTH, IMAGE_HEIGHT);
```
I guess these comments can go, are they still in the fabric code? 🤔

tyra314 commented on src/main/java/hunternif/mc/impl/atlas/client/gui/GuiBookmarkButton.java:
```diff
@@ -56,12 +57,14 @@ public void render(MatrixStack matrices, int mouseX, int mouseY, float partialTi
-		AtlasRenderHelper.drawFullTexture(matrices, iconTexture,
-				getGuiX() + (isMouseOver || isSelected() ? 3 : 2),
-				getGuiY() + 1, ICON_WIDTH, ICON_HEIGHT);
+//		AtlasRenderHelper.drawFullTexture(matrices, iconTexture,
```
I guess these comments can go, are they still in the fabric code? 🤔

tyra314 commented on src/main/java/hunternif/mc/impl/atlas/client/gui/core/GuiBlinkingImage.java:
```diff
@@ -1,19 +1,19 @@
 package hunternif.mc.impl.atlas.client.gui.core;
 
+import org.lwjgl.opengl.GL11;
```
I don't think the import is used

Stereowalker replied:
It's Used Here
GlStateManager.blendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);

tyra314 replied:
My bad. nvm.

tyra314 commented on src/main/java/hunternif/mc/impl/atlas/marker/GlobalMarkersDataHandler.java:
```diff
@@ -56,7 +56,7 @@ public GlobalMarkersData getData() {
 	/** Synchronizes global markers with the connecting client. */
 	public void onPlayerLogin(ServerPlayerEntity player) {
-		data.syncOnPlayer(player);
+		if(data != null) data.syncOnPlayer(player);
```
This is a workaround for #309, which is the same issue as BIOX in discord has, right?

I'm wondering why `data` is `null`. Must have something to do with a different event order in forge?

Stereowalker replied:
Maybe it's a dimension thing. after all `data` is only set during world load when the world loaded is the overworld, so at that point, `data` would be null if for some reason the overworld was not loaded. Though it's just a hypothesis as I've never really used mods that mess with the game that way.

tyra314 commented on src/main/java/hunternif/mc/impl/atlas/mixin/MixinCartographyContainer.java:
```diff
@@ -0,0 +1,95 @@
+import javax.annotation.Nullable;
+
+//TODO Verify if the methods in this yarn methods in this class actually equal the MCP versions with Linkie
+@Mixin(targets = "net.minecraft.inventory.container.CartographyContainer$4")
```
These mixins only worked in dev environment for me. I guess it will break on forge to.
See https://github.com/AntiqueAtlasTeam/AntiqueAtlas/commit/a5614b5c4d7a80148d4f3c7987e175ca265b935b

tyra314 replied:
Apparently, because they try to mixin into an anonymous class, somehow the mapping doesn't work

tyra314 replied:
Yes, you get the same issue when loading the mod outside of a dev env:

> [20:10:59] [pool-3-thread-1/FATAL] [mixin/]: Mixin apply failed antiqueatlas.mixins.json:MixinCartographyContainerSlot -> net.minecraft.inventory.container.CartographyContainer$4: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException Critical injection failure: @ Inject annotation on antiqueatlas_canInsert could not find any targets matching 'isItemValid' in net.minecraft.inventory.container.CartographyContainer$4.

Stereowalker committed:
Just a little update (not yet working)

Stereowalker committed:
Fixing freaking Mixins

Stereowalker committed:
Add markers for Ocean Monument & Woodland Mansion

Stereowalker committed:
Adds new markers for mansion and monuments

Stereowalker committed:
Fixed mipping for end city markers

Stereowalker committed:
Tweaked nether scan radius to be more in line with the visibility range

Stereowalker committed:
Tweaked nether fortress mapping

Stereowalker committed:
Updated mansion marker with the outline

Stereowalker committed:
Adds advancement for the crating recipe so it shows up in the crafting menu

Stereowalker committed:
Adds a new recipe for the empty atlas.

Stereowalker committed:
Fixes SWAMP texture set being overwritten.

Stereowalker committed:
Fixes multiple autoVillage marker for every village

Stereowalker committed:
Some refactoring

Stereowalker committed:
Makes out-of-view markers stick to the edge

Stereowalker committed:
Updated version to try and resolve conflict

tyra314 commented on src/main/java/hunternif/mc/impl/atlas/mixin/MixinCartographyContainer.java:
```diff
@@ -0,0 +1,115 @@
+	@Overwrite
+	private void func_216993_a(ItemStack stack, ItemStack p_216993_2_, ItemStack p_216993_3_) {
+		this.worldPosCallable.consume((p_216996_4_, p_216996_5_) -> {
+			System.out.println("Consuming LOL");
```
Better remove these println in this function :)

tyra314 reviewed (changes requested):
Thanks for keeping up your work. Only minor things

Stereowalker committed:
Move some classes around

Stereowalker committed:
Merge antiqueatlasoverlay into main mod

This replaces the minimap with an more immersive atlas renderer akin to the vanilla map, when hold in the offhand.

Stereowalker committed:
Fix TextureSet of the taiga mountains biome

Stereowalker committed:
Update cloth config and other dependencies

Stereowalker committed:
Renamed SAVANNA_PLATEAU texture set

Stereowalker committed:
Replaces Textures and TextureSets with json files

Stereowalker committed:
Replaces all hard-coded Biome and Tile to TextureSet mappings

Stereowalker committed:
Updated everything in the assets and data folder

Stereowalker committed:
Mainly house-keeping

Moving stuff around, removing old things, renaming for clarity, etc.

Stereowalker committed:
Removes dead code

Stereowalker committed:
Moves location of markers in resource packs

Stereowalker committed:
Improved first-person atlas rendering

The atlas now uses light values, making it the same brightness as maps. This also slightly changes the way how to fix z-fighting, which should fix the shader issues.

Fixes #320, #301, #293, #290

Stereowalker committed:
Fixes NPE during resource pack load

Stereowalker committed:
Fix MarkerApi not usable on server

Stereowalker committed:
Remove unused StructureWatcher stuff.

It has been replaced with stuffs

Stereowalker committed:
Tiny refactoring

Stereowalker committed:
Adds village markers back

Stereowalker committed:
Fixes z-fighting in OverlayRenderer

Stereowalker committed:
Fixes itemNeeded handling in during registration

Stereowalker committed:
Adds handling of missing biome categories

Stereowalker committed:
tiny refactoring

Stereowalker committed:
Makes resource pack loading more robust

Stereowalker committed:
Updated asset and data files

tyra314 reviewed (changes requested):
Code looks good so far, at least what you can see in 337 files 🤷 

But I can't launch the game, neither in dev nor using the build artifact:

```
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: LVT in net/minecraft/client/renderer/FirstPersonRenderer::func_228396_a_(FLcom/mojang/blaze3d/matrix/MatrixStack;Lnet/minecraft/client/renderer/IRenderTypeBuffer$Impl;Lnet/minecraft/client/entity/player/ClientPlayerEntity;I)V has incompatible changes at opcode 151 in callback net/minecraft/client/renderer/FirstPersonRenderer::renderAtlas.
Expected: [F, Lcom/mojang/blaze3d/matrix/MatrixStack;, Lnet/minecraft/client/renderer/IRenderTypeBuffer;, I, Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;, Z, Lnet/minecraft/util/HandSide;]
   Found: [F, Lnet/minecraft/util/Hand;, F, Z, Z, Lnet/minecraft/item/ItemStack;, Lnet/minecraft/item/ItemStack;]

```
Looks like a Mixin problem

Stereowalker committed:
Adds cloth-config2 as required mod dependency

Old and busted: we could jar-in-jar the config stuff.
New hotness: Rely on fabric mod dependencies
Soon™: People being unable to add one mod :(

Co-Authored-By: tyra314 <16997177+tyra314@users.noreply.github.com>

Stereowalker committed:
Working on the API and some internal refactoring

This commit adds some love to the API. I've no idea, if that makes the API usable though. ¯\_(ツ)_/¯
Anyways, I'm using it where I see fit in AA itself.

Co-Authored-By: tyra314 <16997177+tyra314@users.noreply.github.com>

Stereowalker committed:
Fix textures loaded on wrong thread

Co-Authored-By: tyra314 <16997177+tyra314@users.noreply.github.com>

Stereowalker committed:
Improved resource pack loading checks

Co-Authored-By: tyra314 <16997177+tyra314@users.noreply.github.com>

Stereowalker committed:
Adds debug hover text in atlas gui

Co-Authored-By: tyra314 <16997177+tyra314@users.noreply.github.com>

Stereowalker committed:
Fixes a stitching bug in TextureSet

Co-Authored-By: tyra314 <16997177+tyra314@users.noreply.github.com>
